components/raftstore/src/store/fsm/apply.rs

```diff
@@ -1354,7 +1383,15 @@ where
                 }
                 CmdType::IngestSst => {
                     assert!(ctx.kv_wb.is_empty());
-                    self.handle_ingest_sst(&ctx.importer, &ctx.engine, req, &mut ssts)
+                    assert_eq!(requests.len(), 1);
+                    if self.pending_sst.is_none() {
+                        let sst = req.get_ingest_sst().get_sst();
+                        self.handle_ingest(&ctx.importer, sst)?;
+                        ctx.pending_ssts.push(sst.to_owned());
+                        return Ok((RaftCmdResponse::default(), ApplyResult::Yield));
+                    }
+                    ssts.push(self.pending_sst.take().unwrap());
```

Comment:
If there is a pending sst, the current request will be discarded, will not it?

Reply:
The sst of the current request must be the same as the pending sst. Because it is saved here before last yield.